index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
+  function TypeIdentifier(namespace, name, version) {
+    return {
+      namespace: namespace,
+      name: name,
+      version: version
+    };
+  }
+
+  //       parseTypeIdentifier :: String
+  //                           -> TypeIdentifier String? String Number?
+  function parseTypeIdentifier(s) {
+    var parsed = RPARSE.exec(s);
+    // parsed is null if the string was empty, or contained newlines
```

Comment:
Not sure whether to leave this comment in, but since the code is highly implicit I think it's nice to have the original intention around.